### DISCUSSION TITLE
[Bug] HexagonLayer, ColumnLayer, and maybe others fail silently with data array (rather than URL)

### DISCUSSION THREAD

**Discussion by Exr0n** [visgl/deck.gl#7037] [Q&A] [answered]
### Description

When I pass my data to a HexagonLayer or ColumnLayer as a Javascript array of objects, the layer does not appear on the map. If instead I JSON-ify the object, host it, and pass the URL, then the layer works as expected. Is that expected behavior? 

### Flavors

- [X] React
- [ ] Python/Jupyter notebook
- [X] MapboxLayer
- [ ] GoogleMapsOverlay
- [ ] CartoLayer
- [ ] DeckLayer/DeckRenderer for ArcGIS

### Expected Behavior

Be able to pass a data object directly, as in [the HeatmapLayer example](https://github.com/visgl/deck.gl/blob/8.7-release/examples/website/data-filter/app.js).

### Steps to Reproduce

I couldn't figure out how to make a CodePen using React DeckGL. [Here's](https://urbanmap.exr0n.com/data/cities.json) some of the data I was using, and [here's](https://github.com/Exr0nProjects/urbanMap/blob/7508c3555013793846b038fa9228e1f31e4e533d/vis/urbanmap/src/components/UrbanMap.jsx#L94) the relevant line of code. Please let me know if I can provide any other info!

### Environment

- Framework version: deck.gl@8.7.11, react-map-gl@7.0.15
- Browser: Brave 1.37.116 using Chromium 100.0.4896.127
- OS: MacOS 11.6 Big Sur

### Logs

Nothing appeared in the console.

**Comment by Pessimistress** [comment, marked as answer]
Here are all the possible values that `data` accepts:

https://deck.gl/docs/api-reference/core/layer#data

Converting this to a Discussion because it is a basic usage of the library.

To create a React demo you can use https://codesandbox.io.
